scripts/updates/auto-updates.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+set -e -u
+BASEDIR=$(dirname "$(realpath "$0")")
+PKGSDIR="${BASEDIR}/../../packages"
+
+# These variables should be set in environment outside of this script.
+# Build updated packages.
+: "${BUILD_PACKAGES:=false}"
+# Commit changes to Git.
+: "${GIT_COMMIT_PACKAGES:=false}"
+# Push changes to remote.
+: "${GIT_PUSH_PACKAGES:=false}"
+
+if [ -z "${GITHUB_API_TOKEN-}" ]; then
+	echo "You need a Github Personal Access Token be set in variable GITHUB_API_TOKEN."
+	exit 1
+fi
+
+for package in $(ls -1 "${PKGSDIR}"); do
+	unset auto_update_flag project version_regexp termux_version termux_epoch latest_version srcurl
+
+	# Extract some build vars
+	build_vars=$(set +e +u; . "${PKGSDIR}/${package}/build.sh" 2>/dev/null; \
+		echo "auto_update_flag=${TERMUX_PKG_AUTO_UPDATE}"; \
+		echo "termux_version=${TERMUX_PKG_VERSION}"; \
+		echo "srcurl=${TERMUX_PKG_SRCURL}" ;\
+	)
+
+	# Eval build vars to local variables
+	source <(echo "$build_vars")
+
+	if [ "${auto_update_flag}" != "true" ]; then
+		echo "[SKIP] ${package} is not auto updatable."
+		continue
+	fi
+	
+	termux_epoch="$(echo "$termux_version" | cut -d: -f1)"
+	termux_version=$(echo "$termux_version" | cut -d: -f2-)
+	if [ "$termux_version" == "$termux_epoch" ]; then
+		# No epoch set.
+		termux_epoch=""
+	else
+		termux_epoch+=":"
+	fi
+
+	# Transform an actual version to regexp
+	version_regexp=$(echo "${termux_version}" | sed s@[0-9]@\\\\d+@g)
```

Comment:
Are you sure you have read the "github-projects.txt" file and understand the format and purpose of fields?

This regexp is used to convert Git tags into format acceptable by `TERMUX_PKG_VERSION` and you can't recover the tag filtering regexp from the version.

Examples:
* `v1.0.0` - no regexp needed, just stripping the leading `v` is sufficient.
* `prog-1_25_1` - need regexp to extract and convert `1_25_1` into `1.25.1`.
  * https://github.com/cronie-crond/cronie/releases/tag/cronie-1.5.7: version is normalized, but still has prefix
  * https://github.com/curl/curl/releases/tag/curl-7_79_1: prefixed and numbers are separated by underscores.

Reply:
Thanks for your reviewing. the `version_regexp` needs more work, I will try it.